src/main/java/org/red5/server/net/rtmp/codec/RTMPProtocolDecoder.java

```diff
@@ -325,6 +335,8 @@ public Packet decodePacket(RTMPConnection conn, RTMPDecodeState state, IoBuffer
         buf.flip();
         try {
             final IRTMPEvent message = decodeMessage(conn, packet.getHeader(), buf);
+            //we have just decoded full buf, need to advance original IoBuffer
+            in.skip(correction + buf.position());

```

Comment:
I believe this was one of the main issues with packet decoding
original buffer was not advanced, so it was being "re-parsed" with part of header cut-off